### PR TITLE
Redesign: Animation Dropdown 

### LINKED ISSUE #6827: Redesign: Use DropDown from the design-system for the animation effect selection
## Feature Description
Currently, we're using only the `DropDownSelect` from the design-system but not the dropdown itself.
We should switch the custom solution to use what's available in the design system.

See [Figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=6534%3A1889).
There's more in the dedicated [Animation section](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories-Post-Stable?node-id=1923%3A424212) but that might not be relevant for this issue.

Screenshot 2021-03-18 at 19 15 19

### PR DESCRIPTION
**Pull request by BrittanyIRL**
## Context

Building off #6505 which redesigns the animation panel to refactor the custom effect dropdown to use shared materials in design system. 

## Summary

Guts the custom dropdown we were using for animation effects. Should look like this: https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=221%3A13978 

## Relevant Technical Choices

- Moved files that create effect dropdown into their own subdirectory - `/animation` was getting too big. 
- Restructured effect data into background and foreground objects that we can use to map through to create our dropdown options and still have access to all the custom values that each animation needs to preview and be selected. 
- Renamed some props internal to the dropDown's `Menu` to quiet some eslint warnings. Fix prop for `dropDownHeight` so that it can be overwritten. 

## To-do

- There's a side convo happening about how to handle the scrollbar in this dropdown since the designs don't account for it. 

## User-facing changes

Should look the same as in #6505
except i shrank some of the fonts a bit to avoid crowding and flipped the foreground dropdown to open up to give it more room.

https://user-images.githubusercontent.com/10720454/112236399-994ed800-8bfd-11eb-9295-c07e9b5cf2ba.mp4

https://user-images.githubusercontent.com/10720454/112236400-99e76e80-8bfd-11eb-95d7-cdfd5b6532f6.mp4

## Testing Instructions

Verify that selecting animation effects has no regression: 
- Select a foreground element, apply different animations to it with the dropDown. See that dropDown works with both cursor and keyboard. 
- Select background image, apply different animations to it with the dropDown. See that effects are disabled with a tooltip telling you why in some cases. 

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6827

### CODE REVIEW

File: assets/src/edit-story/components/panels/design/animation/effectChooserDropdown/effectChooserDropdown.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+          ...(directions || []).map((dir) => getDirectionalEffect(effect, dir)),
+        ],
+        []
+      );
+    return Object.keys(backgroundEffectOptions).filter((directionalEffect) =>
+      disabledDirectionalEffects.includes(directionalEffect)
+    );
+  }, [disabledTypeOptionsMap]);
+
+  const disabledBackgroundEffects = getDisabledBackgroundEffects();
```
Comment (maxyinger) [inline assets/src/edit-story/components/panels/design/animation/effectChooserDropdown/effectChooserDropdown.js]: should this just be a useMemo now?

File: assets/src/edit-story/components/panels/design/animation/effectChooserDropdown/effectChooserDropdown.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    const selectedAnimation = animationOptionsObject[value].animation;
+    onAnimationSelected({
+      animation: selectedAnimation.value,
+      panDir: selectedAnimation.panDirection,
+      zoomDirection: selectedAnimation.zoomDirection,
+      flyInDir: selectedAnimation.flyInDirection,
+      rotateInDir: selectedAnimation.rotateInDirection,
+      whooshInDir: selectedAnimation.whooshInDirection,
+      scaleDirection: selectedAnimation.scaleDirection,
+    });
```
Comment (maxyinger) [inline assets/src/edit-story/components/panels/design/animation/effectChooserDropdown/effectChooserDropdown.js]: nice. didn't think to just pass everything, but totally works and simplifies things 🎉

File: assets/src/edit-story/components/panels/design/animation/effectChooserDropdown/effectChooserDropdown.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  selectedEffectTitle: PropTypes.string,
+  selectedEffectType: PropTypes.string,
+  onNoEffectSelected: PropTypes.func.isRequired,
+  disabledTypeOptionsMap: PropTypes.objectOf(
+    PropTypes.shape({
+      tooltip: PropTypes.string,
+      options: PropTypes.arrayOf(PropTypes.string),
+    })
+  ),
+};
```
Comment (maxyinger) [inline assets/src/edit-story/components/panels/design/animation/effectChooserDropdown/effectChooserDropdown.js:164]: ~1000 -> 177  👏 👏 👏

**Review by maxyinger — APPROVED** [review]
Looking great in the direction it's heading. Awesome how many lines you've trimmed off of this! bravo 👏

File: assets/src/design-system/components/dropDown/select/components.js [hunk 4]
```diff
@@ -47,10 +47,7 @@ export const SelectButton = styled.button(
     text-overflow: ellipsis;
     cursor: pointer;
 
-    ${themeHelpers.focusableOutlineCSS(
-      theme.colors.border.focus,
-      theme.colors.bg.secondary
-    )}
+    ${themeHelpers.focusableOutlineCSS}
```
Comment (BrittanyIRL) [inline assets/src/design-system/components/dropDown/select/components.js:50]: swapping this back to the default, we need to take a closer look at how to handle other color backgrounds but doing this here affects all of the dropDowns. I'm overriding the one in the animation panel separately below.

File: assets/src/design-system/components/dropDown/select/components.js [hunk 5]
```diff
@@ -86,6 +83,7 @@ export const SelectButton = styled.button(
         color: ${theme.colors.fg.disable};
       }
     }
+    ${selectButtonStylesOverride};
```
Comment (BrittanyIRL) [inline assets/src/design-system/components/dropDown/select/components.js:86]: Added ability to override select button styles since this was part of the original reason the `DropDown` wasn't used for the animation effects.

File: assets/src/design-system/components/menu/list/defaultListItem.js [hunk 6]
```diff
@@ -37,15 +37,15 @@ const ActiveIcon = styled(CheckmarkSmall)`
   transform: translateY(-50%);
 `;
 
-const DefaultListItem = ({ option, isSelected, ...rest }, ref) => (
```
Comment (BrittanyIRL) [inline assets/src/design-system/components/menu/list/defaultListItem.js]: updating the props internal here (`isSelected` is now `selected`, `listLength` is now `listlength` so that we can quiet some of the eslint warnings about naming.

File: assets/src/edit-story/components/panels/design/animation/effectChooserDropdown/effectChooserDropdown.js [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+      const selectedAnimation = animationOptionsObject[value].animation;
+
+      onAnimationSelected({
+        animation: selectedAnimation.value,
+        panDir: selectedAnimation?.panDirection,
+        zoomDirection: selectedAnimation?.zoomDirection,
+        flyInDir: selectedAnimation?.flyInDirection,
+        rotateInDir: selectedAnimation?.rotateInDirection,
+        whooshInDir: selectedAnimation?.whooshInDirection,
+        scaleDirection: selectedAnimation?.scaleDirection,
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/panels/design/animation/effectChooserDropdown/effectChooserDropdown.js:126]: I evened out the naming internal here with `dropDownConstants` but didn't want to rename things outside of this area to avoid scope creep.

File: assets/src/edit-story/components/panels/design/animation/effectChooserDropdown/effectChooserDropdown.js (hunk 2 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/panels/design/animation/effectChooserDropdown/effectChooserDropdown.js]: I played with the idea of checking for these keys and then passing what was found but kind of liked how readable this is to track animation options?

Comment (maxyinger) [reply assets/src/edit-story/components/panels/design/animation/effectChooserDropdown/effectChooserDropdown.js]: Yea completely agree, just lets them all be handled the same. And their creator functions discern arguments they're intaking anyway so no worries there. v nice

**Comment by BrittanyIRL** [comment]
@miina when you're done reviewing will you also add consent? Since I branched off your animation redesign it wants your consent :)

**Comment by miina** [comment]
@googlebot I consent.

**Comment by miina** [comment]
> when you're done reviewing will you also add consent?

Added the consent! Sorry, I was planning to review today but things got in the way -- will do it first thing in the morning.

**Comment by BrittanyIRL** [comment]
> > when you're done reviewing will you also add consent?
> 
> Added the consent! Sorry, I was planning to review today but things got in the way -- will do it first thing in the morning.

know the feeling! no hurry ✨

**Review by maxyinger — APPROVED** [review]
beautiful!

File: assets/src/edit-story/components/panels/design/animation/stories/directionRadioInput.js [hunk 8]
```diff
@@ -30,7 +30,7 @@ import {
 import { DirectionRadioInput } from '../directionRadioInput';
 
 export default {
-  title: 'Animations/Direction Radio Input',
+  title: 'Stories Editor/Components/Panels/Animations/Direction Radio Input',
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/panels/design/animation/stories/directionRadioInput.js:33]: just thought these make more sense now in `Stories Editor/Components` not `Animations`

Comment (maxyinger) [reply assets/src/edit-story/components/panels/design/animation/stories/directionRadioInput.js:33]: yea completely agree. Actually mirrors the file structure in the app. I think if you're looking at the story book implementation now it actually gives you some insight on how to find it in the app. good call

File: assets/src/design-system/components/menu/list/defaultListItem.js (hunk 6 above)
Comment (miina) [reply assets/src/design-system/components/menu/list/defaultListItem.js]: Hm, is this new? The eslint naming warnings for these? Perhaps it's something that would need changing from the config?

File: assets/src/edit-story/components/panels/design/animation/effectInput.js [hunk 9]
```diff
@@ -56,9 +56,9 @@ const StyledInput = styled(NumericInput)`
 const StyledDropDown = styled(DropDown)`
   button {
     height: 46px;
-    ${outerGridBordersCss}
+    ${outerGridBordersCss};
     &:hover {
-      ${outerGridBordersCss}
+      ${outerGridBordersCss};
```
Comment (miina) [inline assets/src/edit-story/components/panels/design/animation/effectInput.js]: Curious about adding the semi-colons to the end of these -- wouldn't this result in double semi-colons since `outerGridBordersCss` already has one as the last character?

**Review by miina — APPROVED** [review]
Awesome work!

File: assets/src/edit-story/components/panels/design/animation/effectInput.js (hunk 9 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/panels/design/animation/effectInput.js]: i was confused by a rebase and trying to figure out why linting was yelling at this, you're right this is unnecessary, i'll remove it!

**Comment by embarks** [comment]
Just a note, in Firefox it looks a bit off-center:
Chrome
Screen Shot 2021-03-26 at 2 12 50 PM
Firefox
Screen Shot 2021-03-26 at 2 12 43 PM

Might be related to scrollbar?

**Review by embarks — APPROVED** [review]
Thank you for breaking up that big file. I know it gets harder to do the bigger they get. And thanks to Max for all his expertise in this area.

**Comment by BrittanyIRL** [comment]
> Might be related to scrollbar?

Indeed! this is absolutely scrollbar related (what's funny is, it's fine in firefox for me - we must have different scrollbar settings on our systems). I know Sam is still thinking through what to do with this.

**Comment by BrittanyIRL** [comment]
(this has a few tests failing,  will fix on monday)

File: assets/src/design-system/components/menu/list/listGroupings.js [hunk 10]
```diff
@@ -64,7 +64,7 @@ const ListGroupings = ({
               !groupOption.disabled && handleMenuItemSelect(event, groupOption)
             }
             option={groupOption}
-            listLength={listLength}
+            $listLength={listLength}
```
Comment (swissspidy) [inline assets/src/design-system/components/menu/list/listGroupings.js:67]: FWIW this does nothing. This prop is not used anywhere.

I'm gonna remove it.

Comment (BrittanyIRL) [reply assets/src/design-system/components/menu/list/listGroupings.js:67]: 🙃  you're right, it doesn't do anything. You can remove it. The reason why it was here is because I was combining the like 4 or 5 dropdowns used across the editor and dashboard and unifying it. At some point in my refactor I decided it was useful to know the length of the list at hand from inside of a list item, which is overridable. Since `listLength` isn't getting passed through via spreading unused props in a potential list item i decided to pass it in anyways as a transient prop so that if we needed it to add a custom style say on the final list item it was super easy.  I added the transient `$` to it here in this Pr because I was cleaning us eslint warnings on the animation dropdown. 

That's why this exists, that said - we don't need this and you are absolutely fine to remove it if you feel like it.
